### PR TITLE
feat: add firewall policy enforcement order

### PR DESCRIPTION
**Pull request by nandagirin**
Hi, this pull request is intended to add parameter to the network module so that we could customize [network_firewall_policy_enforcement_order](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_network#network_firewall_policy_enforcement_order). If you have any concerns and feedbacks, let's discuss it yah, I'll try my best to adjust the PR so it could suit everyone. Thank you 🙏

### CODE REVIEW

**Comment by imrannayer** [comment]
@nandagirin thanks for the PR. Is it possible if you can change variable name to `network_firewall_policy_enforcement_order` just to avoid confusion if this will be applicable on network firewall policy or hierarchical firewall policy.

**Comment by nandagirin** [comment]
> @nandagirin thanks for the PR. Is it possible if you can change variable name to `network_firewall_policy_enforcement_order` just to avoid confusion if this will be applicable on network firewall policy or hierarchical firewall policy.

Sure @imrannayer, I had updated it in the latest commit yah. Thank you

**Comment by daniel-cit** [comment]
Adding the field `network_firewall_policy_enforcement_order`
will also changes the minimum version of the Google provider 
- https://github.com/terraform-google-modules/terraform-google-network/blob/master/versions.tf#L22
- https://github.com/terraform-google-modules/terraform-google-network/blob/master/versions.tf#L26

to [4.60.0 (April 4, 2023)](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4600-april-4-2023)

**Comment by imrannayer** [comment]
@nandagirin Can you plz fix the error. As @daniel-cit mentioned min version required also need to change as this feature was released in 4.60. Better change it to 4.64 cos there was a bug which was fixed in 4.64. This will make it a breaking change. You will need to add an upgrade doc in [docs dir](https://github.com/terraform-google-modules/terraform-google-network/tree/master/docs). It will be similar to [this](https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/blob/main/docs/upgrading_to_v1.0.md)

**Comment by nandagirin** [comment]
Hi @daniel-cit @imrannayer I'm very sorry for the late reply. Thank you so much for the feedback yah. I'll try fixing it soon and make the adjustment needed regarding the docs

**Comment by nandagirin** [comment]
> @nandagirin Can you plz fix the errors. As @daniel-cit mentioned min version required also need to change as this feature was released in 4.60. Better change it to 4.64 cos there was a bug which was fixed in 4.64. This will make it a breaking change. You will need to add an upgrade doc in [docs dir](https://github.com/terraform-google-modules/terraform-google-network/tree/master/docs). It will be similar to [this](https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/blob/main/docs/upgrading_to_v1.0.md)

Hi @imrannayer, regarding the upgrade doc, the existing one there is present with title `upgrading_to_v2.0.md`. Since this PR will introduce breaking changes due to the upgrade of minimum provider version, should we name the upgrade doc as `upgrading_to_v8.0.0.md`, which means the new release will be tagged as v8.0.0?

**Comment by imrannayer** [comment]
> > @nandagirin Can you plz fix the errors. As @daniel-cit mentioned min version required also need to change as this feature was released in 4.60. Better change it to 4.64 cos there was a bug which was fixed in 4.64. This will make it a breaking change. You will need to add an upgrade doc in [docs dir](https://github.com/terraform-google-modules/terraform-google-network/tree/master/docs). It will be similar to [this](https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/blob/main/docs/upgrading_to_v1.0.md)
> 
> Hi @imrannayer, regarding the upgrade doc, the existing one there is present with title `upgrading_to_v2.0.md`. Since this PR will introduce breaking changes due to the upgrade of minimum provider version, should we name the upgrade doc as `upgrading_to_v8.0.0.md`, which means the new release will be tagged as v8.0.0?

yes

**Comment by nandagirin** [comment]
Hi @imrannayer, I've added the fix to the previous error and set the minimum provider version to 4.64 alongside with the upgrade doc in the latest two commits. Please kindly check yah. Thank you

**Comment by imrannayer** [comment]
@nandagirin Can you also update min version needed in vpc sub-module version.tf file.

**Comment by nandagirin** [comment]
Sure @imrannayer I had updated it in the latest commit yah

File: modules/vpc/versions.tf [hunk 1]
```diff
@@ -20,11 +20,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 3.45"
+      version = ">= 4.64"
```
Comment (imrannayer) [inline modules/vpc/versions.tf:23]: change it to 

```
version = "< 5.0, >= 4.64"
```

File: modules/vpc/versions.tf [hunk 2]
```diff
@@ -20,11 +20,11 @@ terraform {
...
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 3.45"
+      version = ">= 4.64"
```
Comment (imrannayer) [inline modules/vpc/versions.tf:27]: change it to 

```
version = "< 5.0, >= 4.64"
```

**Review by imrannayer — APPROVED** [review]

**Pull request merged**